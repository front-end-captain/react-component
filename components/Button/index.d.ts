import * as React from "react";

export declare type ButtonType = "default" | "primary" | "dashed" | "danger";
export declare type ButtonSize = "large" | "default" | "small";
export declare type ButtonHtmlType = "button" | "submit" | "reset";

export interface ButtonProps {
  disabled?: boolean;
  loading?: boolean;
  type?: ButtonType;
  size?: ButtonSize;
  prefixCls?: string;
  className?: string;
  htmlType?: ButtonHtmlType;
  onClick?: React.MouseEventHandler<HTMLAnchorElement>;
  children?: React.ReactNode;
}

declare const Button: (props: ButtonProps) => React.ReactNode;

export default Button;
