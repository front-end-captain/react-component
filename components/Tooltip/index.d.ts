import * as React from "react";

export type TooltipTrigger = "hover" | "click";

export type TooltipPlacement = "top" | "left" | "right" | "bottom";

export interface AbstractTooltipProps {
  prefixCls?: string;
  overlayClassName?: string;
  placement?: TooltipPlacement;
  defaultVisible?: boolean;
  visible?: boolean;
  onVisibleChange?: (visible: boolean) => void;
  trigger?: TooltipTrigger;
  getPopupContainer?: (triggerNode?: HTMLElement) => HTMLElement;
  children?: React.ReactNode;
}

export interface TooltipProps extends AbstractTooltipProps {
  title?: React.ReactNode | RenderFunction;
}

export default class Tooltip extends React.Component<TooltipProps, any> {}
