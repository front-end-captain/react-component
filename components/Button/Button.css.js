import styled from "styled-components";
import { primaryColor, dangerColor, defaultBorderColor, fontSize } from "./../_style/vars.js";

const ButtonWrapper = styled.button`
  line-height: 1.499;
  display: inline-block;
  font-weight: 400;
  text-align: center;
  touch-action: manipulation;
  cursor: pointer;
  outline: none;
  background-image: none;
  border: 1px solid transparent;
  white-space: nowrap;
  padding: 0 15px;
  font-size: ${fontSize};
  border-radius: 4px;
  height: 32px;
  user-select: none;
  transition: all 0.3s cubic-bezier(0.645, 0.045, 0.355, 1);
  position: relative;
  box-shadow: 0 2px 0 rgba(0, 0, 0, 0.015);
  color: rgba(0, 0, 0, 0.65);
  background-color: #fff;
  border-color: ${defaultBorderColor};

  &.${(props) => props.prefixCls}-danger {
    color: ${dangerColor};
    background-color: #f5f5f5;
    border-color: ${defaultBorderColor};
  }

  &.${(props) => props.prefixCls}-danger:hover {
    color: #fff;
    background-color: #ff4d4f;
    border-color: #ff4d4f;
  }

  &.${(props) => props.prefixCls}-primary {
    color: #fff;
    background-color: ${primaryColor};
    border-color: #1890ff;
    text-shadow: 0 -1px 0 rgba(0, 0, 0, 0.12);
    box-shadow: 0 2px 0 rgba(0, 0, 0, 0.045);
  }

  &.${(props) => props.prefixCls}-primary:hover {
    color: #fff;
    background-color: #40a9ff;
    border-color: #40a9ff;
  }

  &.${(props) => props.prefixCls}-dashed {
    color: rgba(0, 0, 0, 0.65);
    background-color: #fff;
    border-color: ${defaultBorderColor};
    border-style: dashed;
  }

  &.${(props) => props.prefixCls}-dashed:hover {
    color: #40a9ff;
    background-color: #fff;
    border-color: #40a9ff;
  }

  &.${(props) => props.prefixCls}[disabled] {
    cursor: not-allowed;
    color: rgba(0, 0, 0, 0.25);
    background-color: #f5f5f5;
    border-color: ${defaultBorderColor};
    text-shadow: none;
    box-shadow: none;
  }
`;

export default ButtonWrapper;
