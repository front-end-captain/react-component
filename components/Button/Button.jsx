import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

import { LoadingIcon } from "./../Icon/index.jsx";

import ButtonWrapper from "./Button.css.js";

const sizes = { small: "small", default: "default", large: "large" };

const types = { primary: "primary", default: "default", danger: "danger", dashed: "dashed" };

const htmlTypes = { button: "button", submit: "submit", reset: "reset" };

const Button = ({
  children,
  className,
  onClick,
  disabled,
  loading,
  htmlType,
  type,
  size,
  prefixCls,
  block,
  ...resetProps
}) => {
  const isDisabled = disabled || loading ? { disabled: true } : { onClick };
  const baseProps = {
    ...resetProps,
    ...isDisabled,
    prefixCls,
    type: htmlType,
    className: classNames(prefixCls, className, {
      [`${prefixCls}-${type}`]: type,
      [`${prefixCls}-default`]: !disabled && type === types.default,
      [`${prefixCls}-normal`]: type === types.default,
      [`${prefixCls}-disabled`]: disabled,
      [`${prefixCls}-loading`]: loading,
      [`${prefixCls}-size-${size}`]: size !== sizes.default,
      [`${prefixCls}-block`]: block,
    }),
  };
  return (
    <ButtonWrapper {...baseProps}>
      {loading && <LoadingIcon className="sky-loading" />}
      <span>{children || "Button"}</span>
    </ButtonWrapper>
  );
};

Button.defaultProps = {
  prefixCls: "sky-btn",
  loading: false,
  type: types.default,
  size: sizes.default,
  htmlType: htmlTypes.button,
  disabled: false,
  block: false,
};

export const ButtonPropTypes = {
  prefixCls: PropTypes.string,
  loading: PropTypes.bool,
  type: PropTypes.oneOf(Object.values(types)),
  size: PropTypes.oneOf(Object.values(sizes)),
  htmlType: PropTypes.oneOf(Object.values(htmlTypes)),
  disabled: PropTypes.bool,
  block: PropTypes.bool,
};

Button.propTypes = ButtonPropTypes;

export default Button;
