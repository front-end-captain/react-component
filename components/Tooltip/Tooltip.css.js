import styled from "styled-components";

import { defaultBoxShadow } from "./../_style/vars.js";
import {
  skySlideBottom,
  skySlideBottomOut,
  skySliderUp,
  skySliderUpOut,
  skySliderLeft,
  skySliderLeftOut,
  skySliderRight,
  skySliderRightOut,
} from "./../_style/animate.js";

const TooltipWrapper = styled.div`
  display: inline-block;
  vertical-align: top;
`;

const OverlayContent = styled.div`
  position: absolute;
  background-clip: border-box;
  box-shadow: ${defaultBoxShadow};
  color: #fff;
  padding: 8px 15px;
  font-size: 14px;
  word-break: break-all;
  z-index: 999;
  transform: scale(0);
  background-color: #555;
  border-radius: 4px;

  &:after,
  &:before {
    content: " ";
    position: absolute;
    width: 0;
    height: 0;
    border: 10px solid transparent;
    z-index: -1;
  }

  &.${(props) => props.prefixCls}-position-top {
    margin-top: -10px;

    &:after,
    &:before {
      left: 50%;
      top: 85%;
      border-top-color: #555;
      border-bottom: none;
      transform: translateX(-50%);
    }
  }

  &.${(props) => props.prefixCls}-position-bottom {
    margin-top: 10px;

    &:after,
    &:before {
      left: 50%;
      bottom: 85%;
      border-top: none;
      border-bottom-color: #555;
      transform: translateX(-50%);
    }
  }

  &.${(props) => props.prefixCls}-position-left {
    margin-left: -10px;

    &:after,
    &:before {
      transform: translateY(-50%);
      top: 50%;
      border-right: none;
      border-left-color: #555;
      left: 95%;
    }
  }

  &.${(props) => props.prefixCls}-position-right {
    margin-left: 10px;

    &:after,
    &:before {
      transform: translateY(-50%);
      top: 50%;
      border-left: none;
      border-right-color: #555;
      right: 95%;
    }
  }

  &.${(props) => props.prefixCls}-not-display {
    margin-left: -10000px;
  }

  &.${(props) => props.prefixCls}-show {
    &.${(props) => props.prefixCls}-position-top {
      animation: ${skySlideBottom} 0.5s ease-in-out forwards;
    }

    &.${(props) => props.prefixCls}-position-bottom {
      animation: ${skySliderUp} 0.5s ease-in-out forwards;
    }

    &.${(props) => props.prefixCls}-position-right {
      animation: ${skySliderRight} 0.5s ease-in-out forwards;
    }

    &.${(props) => props.prefixCls}-position-left {
      animation: ${skySliderLeft} 0.5s ease-in-out forwards;
    }
  }

  &.${(props) => props.prefixCls}-hide {
    opacity: 0;
    pointer-events: none;

    &.${(props) => props.prefixCls}-position-top {
      animation: ${skySlideBottomOut} 0.5s ease-in-out forwards;
    }

    &.${(props) => props.prefixCls}-position-bottom {
      animation: ${skySliderUpOut} 0.5s ease-in-out forwards;
    }

    &.${(props) => props.prefixCls}-position-right {
      animation: ${skySliderRightOut} 0.5s ease-in-out forwards;
    }

    &.${(props) => props.prefixCls}-position-left {
      animation: ${skySliderLeftOut} 0.5s ease-in-out forwards;
    }
  }
`;

const TriggerWrapper = styled.div`
  display: inline-block;
  cursor: pointer;
`;

export { TooltipWrapper, OverlayContent, TriggerWrapper };
