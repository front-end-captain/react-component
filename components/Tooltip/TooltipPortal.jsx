import React from "react";
import { createPortal } from "react-dom";

// props
// getPopupContainer: () => void;
// onChange: () => void;

class TooltipPortal extends React.Component {
  static defaultProps = {
    getPopupContainer: () => document.body,
  };

  constructor(props) {
    super(props);

    this.wrapperNode = document.createElement("div");
    this.wrapperContainer = props.getPopupContainer();
  }

  componentDidMount() {
    const { onChange } = this.props;
    this.wrapperContainer.appendChild(this.wrapperNode);
    this.forceUpdate();
    if (onChange) {
      onChange();
    }
  }

  componentWillUnmount() {
    this.wrapperContainer.removeChild(this.wrapperNode);
  }

  render() {
    const { children } = this.props;
    return createPortal(children, this.wrapperNode);
  }
}

export default TooltipPortal;
