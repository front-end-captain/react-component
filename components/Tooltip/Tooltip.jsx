import React, { createRef } from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import scrollIntoViewIfNeeded from "scroll-into-view-if-needed";

import TooltipPortal from "./TooltipPortal.jsx";

import { OverlayContent, TooltipWrapper, TriggerWrapper } from "./Tooltip.css.js";

class Tooltip extends React.Component {
  static defaultProps = {
    prefixCls: "sky-tooltip",
    placement: "top",
    onVisibleChange: () => {},
    trigger: "hover",
  };

  static propTypes = {
    prefixCls: PropTypes.string,
    className: PropTypes.string,
    placement: PropTypes.oneOf(["top", "right", "left", "bottom"]),
    onVisibleChange: PropTypes.func,
    trigger: PropTypes.oneOf(["hover", "click"]),
  };

  static triggerTypes = { hover: "hover", click: "click" };

  OPEN_LOCK = false;

  CLOSE_LOCK = false;

  constructor(props) {
    super(props);

    this.state = {
      visible: null,
      left: 0,
      top: 0,
    };

    this.overlayContent = createRef();
    this.trigger = createRef();
    this.wrapper = createRef();

    this.onCloseTooltip = this.onCloseTooltip.bind(this);
    this.onOpenTooltip = this.onOpenTooltip.bind(this);
    this.getWrapperBounding = this.getWrapperBounding.bind(this);
    this.setWrapperBounding = this.setWrapperBounding.bind(this);
    this.resetWrapperPosition = this.resetWrapperPosition.bind(this);
    this.onClickOutSideHandle = this.onClickOutSideHandle.bind(this);
  }

  componentDidMount() {
    document.addEventListener("click", this.onClickOutSideHandle);
  }

  componentWillReceiveProps(nextProps) {
    this.setState(
      {
        visible: nextProps.visible,
      },
      () => {
        this.OPEN_LOCK = !nextProps.visible;
        this.CLOSE_LOCK = !nextProps.visible;
      },
    );
  }

  componentWillUnmount() {
    document.removeEventListener("click", this.onClickOutSideHandle);
  }

  onClickOutSideHandle(event) {
    event.stopPropagation();

    const { visible } = this.state;
    const { onVisibleChange } = this.props;

    if (
      visible &&
      !this.wrapper.current.contains(event.target) &&
      !this.overlayContent.current.contains(event.target)
    ) {
      this.setState({ visible: false }, () => {
        this.OPEN_LOCK = false;
        this.CLOSE_LOCK = true;
        onVisibleChange(false);
      });
    }
  }

  onOpenTooltip() {
    const { onVisibleChange } = this.props;
    this.CLOSE_LOCK = false;
    this.setState({ visible: true }, () => {
      if (!this.OPEN_LOCK) {
        this.setWrapperBounding(() => {
          onVisibleChange(true);
          this.OPEN_LOCK = true;
          this.CLOSE_LOCK = false;
          scrollIntoViewIfNeeded(this.overlayContent.current, {
            scrollMode: "if-needed",
            behavior: "smooth",
            block: "nearest",
            inline: "nearest",
          });
        });
      }
    });
  }

  onCloseTooltip() {
    const { onVisibleChange } = this.props;
    this.setState({ visible: false }, () => {
      this.setWrapperBounding(() => {
        onVisibleChange(false);
        this.OPEN_LOCK = false;
        this.CLOSE_LOCK = true;
      });
    });
  }

  getWrapperBounding() {
    const { scrollX, scrollY } = window;
    const { width, height, top, left } = this.trigger.current.getBoundingClientRect();
    const {
      height: wrapperHeight,
      width: wrapperWidth,
    } = this.overlayContent.current.getBoundingClientRect();

    const { placement } = this.props;

    const positions = {
      top: {
        top: top + scrollY - wrapperHeight,
        left: left + scrollX + width / 2 - wrapperWidth / 2,
      },
      bottom: {
        top: top + height + scrollY,
        left: left + scrollX + width / 2 - wrapperWidth / 2,
      },
      left: {
        top: top + scrollY + height / 2 - wrapperHeight / 2,
        left: left + scrollX - wrapperWidth,
      },
      right: {
        top: top + scrollY + height / 2 - wrapperHeight / 2,
        left: left + scrollX + width,
      },
    };

    return positions[placement];
  }

  setWrapperBounding(cb = () => {}) {
    const { left, top } = this.getWrapperBounding();
    this.setState({ left, top }, cb);
  }

  resetWrapperPosition() {
    const { visible } = this.state;
    if (visible) {
      setTimeout(() => {
        this.setWrapperBounding();
      }, 0);
    }
  }

  render() {
    const { left, top, visible } = this.state;
    const {
      children,
      title,
      prefixCls,
      className,
      placement,
      getPopupContainer,
      overlayClassName,
      trigger,
    } = this.props;

    const contentClassNames = classNames(
      overlayClassName,
      `${prefixCls}-overlay`,
      `${prefixCls}-position-${placement}`,
      {
        [`${prefixCls}-hide`]: !visible,
        [`${prefixCls}-show`]: visible,
        [`${prefixCls}-not-display`]: this.trigger.current === null,
      },
    );

    const isHover = trigger === Tooltip.triggerTypes.hover;

    const triggerEvents = isHover
      ? { onMouseEnter: this.onOpenTooltip, onMouseLeave: this.onCloseTooltip }
      : { onClick: this.onOpenTooltip };

    return (
      <TooltipWrapper
        className={classNames(prefixCls, className)}
        ref={this.wrapper}
        {...triggerEvents}
      >
        <TooltipPortal onChange={this.resetWrapperPosition} getPopupContainer={getPopupContainer}>
          <OverlayContent
            ref={this.overlayContent}
            className={contentClassNames}
            style={{ left, top }}
            prefixCls={prefixCls}
          >
            {title}
          </OverlayContent>
        </TooltipPortal>

        <TriggerWrapper ref={this.trigger}>{children}</TriggerWrapper>
      </TooltipWrapper>
    );
  }
}

export default Tooltip;
