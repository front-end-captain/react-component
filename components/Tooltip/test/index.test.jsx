import React from "react";
import assert from "power-assert";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Tooltip from "./../index.js";
import TooltipPortal from "./../TooltipPortal.jsx";
import Button from "./../../Button/index.js";

describe("Tooltip", () => {
  it("should render a <Tooltip> component", () => {
    const wrapper = render(
      <div>
        <Tooltip title="prompt text">put the mouse on this</Tooltip>
        <Tooltip title="prompt text" placement="top">
          <Button>top</Button>
        </Tooltip>
        <Tooltip title="prompt text" placement="right">
          <Button>right</Button>
        </Tooltip>
        <Tooltip title="prompt text" placement="bottom">
          <Button>bottom</Button>
        </Tooltip>
        <Tooltip title="prompt text" placement="left">
          <Button>left</Button>
        </Tooltip>
      </div>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should find `.sky-tooltip` classnames", () => {
    const wrapper = shallow(<Tooltip title="prompt text">put the mouse on this</Tooltip>);
    assert(wrapper.find(".sky-tooltip").length === 1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render specific positions", () => {
    const wrapper = shallow(
      <div>
        <Tooltip title="prompt text" placement="top">
          <Button>top</Button>
        </Tooltip>
        <Tooltip title="prompt text" placement="right">
          <Button>right</Button>
        </Tooltip>
        <Tooltip title="prompt text" placement="bottom">
          <Button>bottom</Button>
        </Tooltip>
        <Tooltip title="prompt text" placement="left">
          <Button>left</Button>
        </Tooltip>
      </div>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should find position classnames", () => {
    const wrapper = shallow(
      <Tooltip placement="top" title="prompt text">
        <Button>top</Button>
      </Tooltip>,
    );

    assert(wrapper.find(".sky-tooltip-position-top").length === 1);
  });

  it("should find TooltipPortal", () => {
    const wrapper = shallow(
      <Tooltip placement="top">
        <Button>top</Button>
      </Tooltip>,
    );

    assert(wrapper.find(TooltipPortal).length === 1);
  });

  it("should find TooltipPortal when visible change", () => {
    const wrapper = shallow(
      <Tooltip placement="top">
        <Button>top</Button>
      </Tooltip>,
    );

    wrapper.setState({ visible: true });
    assert(wrapper.find(TooltipPortal).length === 1);

    wrapper.setState({ visible: false });
    assert(wrapper.find(TooltipPortal).length === 1);
  });

  it("should find custom children", () => {
    const wrapper = shallow(
      <Tooltip placement="top">
        <Button>top</Button>
      </Tooltip>,
    );

    assert(wrapper.find(Button).length === 1);
  });

  it("should invoking externals callback: onVisibleChange ", () => {
    const onChange = jest.fn();

    const wrapper = shallow(
      <Tooltip placement="top" onVisibleChange={onChange} title="prompt text">
        <Button>top</Button>
      </Tooltip>,
    );

    wrapper.find(".sky-tooltip-position-top").simulate("mouseleave");

    setTimeout(() => {
      expect(onChange).toHaveBeenCalled();
    }, 200);
  });

  it("should contain children by TooltipPortal", () => {
    const wrapper = shallow(
      <TooltipPortal>
        <Button>按钮</Button>
      </TooltipPortal>,
    );

    expect(wrapper.find(Button)).toHaveLength(1);
  });

  it("should append div#test to body by TooltipPortal", () => {
    const wrapper = shallow(
      <TooltipPortal>
        <div id="test">test</div>
      </TooltipPortal>,
    );

    expect(wrapper.find("#test")).toHaveLength(1);
  });

  it("should remove div#test when TooltipPortal un mount", () => {
    const wrapper = shallow(
      <TooltipPortal>
        <div id="test">test</div>
      </TooltipPortal>,
    );

    wrapper.unmount();
    expect(wrapper.find("#test")).toHaveLength(0);
  });

  it("window can not call click event handle when tooltip un mount", () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <Tooltip title="prompt text">
        <Button>按钮</Button>
      </Tooltip>,
    );

    wrapper.unmount();

    window.onclick = () => onClick;

    expect(onClick).not.toHaveBeenCalled();
  });

  it("should update visible state when props visible change", () => {
    const onChange = jest.fn();

    const wrapper = shallow(
      <Tooltip title="prompt text" onVisibleChange={onChange}>
        <Button>按钮</Button>
      </Tooltip>,
    );

    wrapper.setProps({ visible: true });
    expect(wrapper.state().visible).toEqual(true);

    wrapper.setProps({ visible: false });
    expect(wrapper.state().visible).toEqual(false);

    expect(onChange).not.toHaveBeenCalled();
  });
});
