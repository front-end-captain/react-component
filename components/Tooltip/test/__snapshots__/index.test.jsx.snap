// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip should find \`.sky-tooltip\` classnames 1`] = `
<Tooltipcss__TooltipWrapper
  className="sky-tooltip"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <TooltipPortal
    getPopupContainer={[Function]}
    onChange={[Function]}
  >
    <Tooltipcss__OverlayContent
      className="sky-tooltip-overlay sky-tooltip-position-top sky-tooltip-hide sky-tooltip-not-display"
      prefixCls="sky-tooltip"
      style={
        Object {
          "left": 0,
          "top": 0,
        }
      }
    >
      prompt text
    </Tooltipcss__OverlayContent>
  </TooltipPortal>
  <Tooltipcss__TriggerWrapper>
    put the mouse on this
  </Tooltipcss__TriggerWrapper>
</Tooltipcss__TooltipWrapper>
`;

exports[`Tooltip should render a <Tooltip> component 1`] = `
<div>
  <div
    class="sky-tooltip Tooltipcss__TooltipWrapper-be2ocw-0 hKKFDA"
  >
    <div
      class="sky-tooltip-overlay sky-tooltip-position-top sky-tooltip-hide sky-tooltip-not-display Tooltipcss__OverlayContent-be2ocw-1 craHJN"
      style="left:0;top:0"
    >
      prompt text
    </div>
    <div
      class="Tooltipcss__TriggerWrapper-be2ocw-2 jUvocl"
    >
      put the mouse on this
    </div>
  </div>
  <div
    class="sky-tooltip Tooltipcss__TooltipWrapper-be2ocw-0 hKKFDA"
  >
    <div
      class="sky-tooltip-overlay sky-tooltip-position-top sky-tooltip-hide sky-tooltip-not-display Tooltipcss__OverlayContent-be2ocw-1 craHJN"
      style="left:0;top:0"
    >
      prompt text
    </div>
    <div
      class="Tooltipcss__TriggerWrapper-be2ocw-2 jUvocl"
    >
      <button
        class="sky-btn sky-btn-default sky-btn-normal Buttoncss__ButtonWrapper-k9s1tq-0 cRNwTE"
        type="button"
      >
        <span>
          top
        </span>
      </button>
    </div>
  </div>
  <div
    class="sky-tooltip Tooltipcss__TooltipWrapper-be2ocw-0 hKKFDA"
  >
    <div
      class="sky-tooltip-overlay sky-tooltip-position-right sky-tooltip-hide sky-tooltip-not-display Tooltipcss__OverlayContent-be2ocw-1 craHJN"
      style="left:0;top:0"
    >
      prompt text
    </div>
    <div
      class="Tooltipcss__TriggerWrapper-be2ocw-2 jUvocl"
    >
      <button
        class="sky-btn sky-btn-default sky-btn-normal Buttoncss__ButtonWrapper-k9s1tq-0 cRNwTE"
        type="button"
      >
        <span>
          right
        </span>
      </button>
    </div>
  </div>
  <div
    class="sky-tooltip Tooltipcss__TooltipWrapper-be2ocw-0 hKKFDA"
  >
    <div
      class="sky-tooltip-overlay sky-tooltip-position-bottom sky-tooltip-hide sky-tooltip-not-display Tooltipcss__OverlayContent-be2ocw-1 craHJN"
      style="left:0;top:0"
    >
      prompt text
    </div>
    <div
      class="Tooltipcss__TriggerWrapper-be2ocw-2 jUvocl"
    >
      <button
        class="sky-btn sky-btn-default sky-btn-normal Buttoncss__ButtonWrapper-k9s1tq-0 cRNwTE"
        type="button"
      >
        <span>
          bottom
        </span>
      </button>
    </div>
  </div>
  <div
    class="sky-tooltip Tooltipcss__TooltipWrapper-be2ocw-0 hKKFDA"
  >
    <div
      class="sky-tooltip-overlay sky-tooltip-position-left sky-tooltip-hide sky-tooltip-not-display Tooltipcss__OverlayContent-be2ocw-1 craHJN"
      style="left:0;top:0"
    >
      prompt text
    </div>
    <div
      class="Tooltipcss__TriggerWrapper-be2ocw-2 jUvocl"
    >
      <button
        class="sky-btn sky-btn-default sky-btn-normal Buttoncss__ButtonWrapper-k9s1tq-0 cRNwTE"
        type="button"
      >
        <span>
          left
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Tooltip should render specific positions 1`] = `
<div>
  <Tooltip
    onVisibleChange={[Function]}
    placement="top"
    prefixCls="sky-tooltip"
    title="prompt text"
    trigger="hover"
  >
    <Button
      block={false}
      disabled={false}
      htmlType="button"
      loading={false}
      prefixCls="sky-btn"
      size="default"
      type="default"
    >
      top
    </Button>
  </Tooltip>
  <Tooltip
    onVisibleChange={[Function]}
    placement="right"
    prefixCls="sky-tooltip"
    title="prompt text"
    trigger="hover"
  >
    <Button
      block={false}
      disabled={false}
      htmlType="button"
      loading={false}
      prefixCls="sky-btn"
      size="default"
      type="default"
    >
      right
    </Button>
  </Tooltip>
  <Tooltip
    onVisibleChange={[Function]}
    placement="bottom"
    prefixCls="sky-tooltip"
    title="prompt text"
    trigger="hover"
  >
    <Button
      block={false}
      disabled={false}
      htmlType="button"
      loading={false}
      prefixCls="sky-btn"
      size="default"
      type="default"
    >
      bottom
    </Button>
  </Tooltip>
  <Tooltip
    onVisibleChange={[Function]}
    placement="left"
    prefixCls="sky-tooltip"
    title="prompt text"
    trigger="hover"
  >
    <Button
      block={false}
      disabled={false}
      htmlType="button"
      loading={false}
      prefixCls="sky-btn"
      size="default"
      type="default"
    >
      left
    </Button>
  </Tooltip>
</div>
`;
